operator/docs/enhancements/zone_aware_replication.md

```diff
@@ -0,0 +1,460 @@
+A zone represents a logical failure domain. It is common for Kubernetes clusters to span multiple zones for increased availability. While the exact definition of a zone is left to infrastructure implementations, common properties of a zone include very low network latency within a zone, no-cost network traffic within a zone, and failure independence from other zones.
+The following sections describe a set of APIs in form of Custom Resource Definitions (CRD) that enable users of `LokiStack` resources to support:
+
+- Enable Zone-Aware Replication configuration in LokiStack CR so the components are deployed in different zones.
+- Configure Loki to span data replication across multiple zones.
+
+## Motivation
+
+Zone-aware replication is the replication of data across failure domains. Avoiding data loss during a domain outage is the motivation to introduce a zone-aware component deployment and enable Loki's zone-aware data replication capabilities.
+
+### Goals
+
+- The user can enable zone aware replication in the loki operator
+- The LokiStack administrator can enable zone-aware data replication for a managed Loki cluster.
+- The LokiStack administrator can choose at least one topology label to enable the cluster spread across domains
+
+### Non-Goals
+
+- Minimize cross-zone bandwidth costs.
+- Introduce data replication across Loki clusters in different regions.
+
+## Proposal
+
+The following enhancement proposal describes the required API additions and changes in the Loki Operator to add zone-aware data replication support.
+
+### API Extensions
+
+#### LokiStack Changes: Support for configuring zone-aware data replication
+
+The following API changes introduce a new spec to enable configuration for Loki's data replication properties.
+
+_Note:_ The new `replicationSpec` introduces in addition a field `factor` that is a replacement for the old `replicationFactor` field. Moving forward the old field is officially deprecated any will be removed in future CRD versions.
+
+```go
+
+import "github.com/prometheus/prometheus/model/labels"
+
+
+//  LokiStackSpec defines the desired state of LokiStack
+type LokiStackSpec struct {
+...
+    // ReplicationFactor defines the policy for log stream replication. (Deprecated: Please use replication.factor instead. This field will be removed in future versions of this CRD)
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:validation:Minimum:=1
```

Comment:
I think "optional" and "minimum=1" will clash, because there is no possibility to set "no value".